app/gateway/proxy.py

```diff
@@ -16,43 +16,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Gateway logic."""
-
 import logging
 import importlib
 import json
 import jwt
-from flask import request, Response
+import re
+from quart import request, Response
 from urllib.parse import urljoin
 
 from .. import app
-from flask_cors import CORS
+from app.auth.web import COOKIE_DOMAIN, COOKIE_SECURE, get_refreshed_tokens
 
 
```

Comment:
CORS is no longer needed?

Reply:
no, it is now at the app root level

Reply:
Actually it could be removed there too. The more restrictive approach to token handling between UI and gateway discussed [here](https://github.com/SwissDataScienceCenter/renku-ui/issues/307) relies on UI and gateway to be running under the same host name. We therefore do not do any cross origin resource sharing (CORS) anymore, and we therefore don't have to support it on the server side anymore.